javascript/javascript1/week2/fullName.js

```diff
@@ -0,0 +1,24 @@
+//Flight booking getFullname function with use of formals:
+function getFullname(firstname, surname, useFormalName, isMistress) {
+  if (firstname === "" && surname === "") {
+    return "Please enter your firstname and surname";
+  } else if (useFormalName) {
+    if (isMistress) {
+      return `Mistress ${firstname} ${surname}`;
+    } else {
+      return `Lord ${firstname} ${surname}`;
+    }
+  } else {
+    return `${firstname} ${surname}`;
+  }
+}
+
+const fullName1 = getFullname("Rajesh", "Bhatt", false, false); //Case for without Formal Name
+const fullName2 = getFullname("Benjamin", "Hughes", true, false); //Case for lord
+const fullName3 = getFullname("Joan", "Nielsen", true, true); //Case for mistress
```

Comment:
What if the 4rth argument is not set? `getFullname("Joan", "Nielsen", true);`? Then your code will return "Lord Joan Nielsen". I'd recommend that you update the logic so if `useFormalName` is set, but `isMistress` is undefined (neither true, nor false) that you would fall back to the simple greeting with only name.

Reply:
I have done it...